File: Python/Mahasiswa.py
from Civitas import Civitas
from Prodi import Prodi
from Course import Course

class Mahasiswa(Civitas):
    # mahasiswa adalah child dari civitas
    
    # atribut private
    __nim = ""
    __prodiA = Prodi()
    __courseB = [Course()]
    
    #constructor
    def __init__(self, nik="", name="", gender="", univ="", email="", fakultas = "", nim = "", prodiA = Prodi(), courseB = [Course()]):
        super().__init__(nik, name, gender, univ, email, fakultas)
        self.__nim = nim
        self.__prodiA = prodiA
        self.__courseB = courseB
    
    # getter setter
    def setNim(self, nim):
        self.__nim = nim
    
    def getNim(self):
        return self.__nim
    
    def setProdi(self, prodiA):
        self.__prodiA = prodiA
    
    def getProdi(self):
        return self.__prodiA

    def setCourse(self, courseB):
        self.__courseB = courseB
    
    def getCourse(self):
        return self.__courseB
    
    def printMhs(self):
        print("NIK              : " + str(self.getNik()))
        print("Nama             : " + str(self.getName()))
        print("Gender           : " + str(self.getGender()))
        print("Universitas      : " + str(self.getUniv()))
        print("Email            : " + str(self.getEmail()))
        print("Fakultas         : " + str(self.getFakultas()))
        print("NIM              : " + str(self.getNim()))
        print("Prodi            : " + str(self.getProdi().getNamaProdi()))
        print("Matkul Kontrak   : " )
        for i in range (len(self.__courseB)):
            print(str(i+1) + ") ", end="")
            print((self.__courseB[i].getNamaMatkul()))

File: CPP/Program/Mahasiswa.cpp
#include "header.hh"

// Constructor
Mahasiswa::Mahasiswa() : Civitas()
{
    nim = "";
}

// Construction dengan atribut
Mahasiswa::Mahasiswa(string nik, string name, string gender, string univ, string email, string fakultas, string nim, Prodi prodiA, Course courseB) : Civitas(nik, name, gender, univ, email, fakultas)
{
    this->nim = nim;
    this->prodiA = prodiA;
}

// Getter Setter
string Mahasiswa::getNIM()
{
    return nim;
}

void Mahasiswa::setNIM(string nim)
{
    this->nim = nim;
}

Prodi Mahasiswa::getProdiA()
{
    return prodiA;
}

void Mahasiswa::setProdiA(Prodi prodiA)
{
    this->prodiA = prodiA;
}

list<Course> Mahasiswa::getCourseB()
{
    return courseB;
}

void Mahasiswa::setCourseB(Course courseB)
{
    this->courseB.push_back(courseB);
}

// Destructor
Mahasiswa::~Mahasiswa() {}